apis/apps/v1/cluster_types.go

```diff
@@ -542,33 +542,44 @@ type ClusterComponentService struct {
 	PodService *bool `json:"podService,omitempty"`
 }
 
-// ShardingSpec defines how KubeBlocks manage dynamic provisioned shards.
+// ClusterSharding defines how KubeBlocks manage dynamic provisioned shards.
 // A typical design pattern for distributed databases is to distribute data across multiple shards,
 // with each shard consisting of multiple replicas.
 // Therefore, KubeBlocks supports representing a shard with a Component and dynamically instantiating Components
 // using a template when shards are added.
 // When shards are removed, the corresponding Components are also deleted.
-type ShardingSpec struct {
+type ClusterSharding struct {
 	// Represents the common parent part of all shard names.
+	//
 	// This identifier is included as part of the Service DNS name and must comply with IANA service naming rules.
-	// It is used to generate the names of underlying Components following the pattern `$(shardingSpec.name)-$(ShardID)`.
+	// It is used to generate the names of underlying Components following the pattern `$(clusterSharding.name)-$(ShardID)`.
 	// ShardID is a random string that is appended to the Name to generate unique identifiers for each shard.
 	// For example, if the sharding specification name is "my-shard" and the ShardID is "abc", the resulting Component name
 	// would be "my-shard-abc".
 	//
-	// Note that the name defined in Component template(`shardingSpec.template.name`) will be disregarded
-	// when generating the Component names of the shards. The `shardingSpec.name` field takes precedence.
+	// Note that the name defined in Component template(`clusterSharding.template.name`) will be disregarded
+	// when generating the Component names of the shards. The `clusterSharding.name` field takes precedence.
 	//
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:MaxLength=15
 	// +kubebuilder:validation:Pattern:=`^[a-z0-9]([a-z0-9\.\-]*[a-z0-9])?$`
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="name is immutable"
 	Name string `json:"name"`
 
+	// Specifies the ShardingDefinition custom resource (CR) that defines the sharding's characteristics and behavior.
+	//
+	// The full name or regular expression is supported to match the ShardingDefinition.
+	//
+	// +kubebuilder:validation:MaxLength=64
+	// +kubebuilder:validation:Pattern:=`^[a-z]([a-z0-9\.\-]*[a-z0-9])?$`
+	// +optional
+	ShardingDef string `json:"shardingDef,omitempty"`
```

Comment:
What is the relationship between compDef in the referenced shardingDef object and template.componentDef? Should it be explained in the API?